show-viewshed-from-point-in-scene/src/main/java/com/esri/arcgismaps/sample/showviewshedfrompointinscene/screens/ViewshedOptionsScreen.kt

```diff
@@ -0,0 +1,277 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showviewshedfrompointinscene.screens
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Checkbox
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
+
+/**
+ * Viewshed options screen for sliders and checkboxes
+ */
+@Composable
+fun ViewshedOptionsScreen(
+    onHeadingChanged: (Float) -> Unit = {},
+    onPitchChanged: (Float) -> Unit = {},
+    onHorizontalAngleChanged: (Float) -> Unit = {},
+    onVerticalAngleChanged: (Float) -> Unit = {},
+    onMinDistanceChanged: (Float) -> Unit = {},
+    onMaxDistanceChanged: (Float) -> Unit = {},
+    isFrustumVisible: (Boolean) -> Unit = {},
+    isAnalysisVisible: (Boolean) -> Unit = {}
+) {
+    Column() {
+        // sliders
+        HeadingSlider(onHeadingChanged)
+        PitchSlider(onPitchChanged)
+        HorizontalAngleSlider(onHorizontalAngleChanged)
+        VerticalAngleSlider(onVerticalAngleChanged)
+        MinimumDistanceSlider(onMinDistanceChanged)
+        MaximumDistanceSlider(onMaxDistanceChanged)
+        // checkbox
+        Row {
+            FrustumCheckBox(isFrustumVisible)
+            AnalysisCheckBox(isAnalysisVisible)
+        }
+    }
+}
+
+@Composable
+private fun HeadingSlider(onHeadingChanged: (Float) -> Unit) {
+
+    var sliderValue by remember {
+        mutableStateOf(82f)
+    }
+    Row {
+        Text(modifier = Modifier.padding(start = 10.dp, top = 10.dp, end = 10.dp).width(150.dp),
+            text = "Heading")
+        Slider(
+            modifier = Modifier.weight(1f),
+            value = sliderValue,
+            onValueChange = {
+                sliderValue = it
+                // update view model viewshed value
+                onHeadingChanged(sliderValue)
+            },
+            valueRange = 0f..360f
+        )
+        Text(
+            modifier = Modifier.padding(start = 10.dp, top = 10.dp, end = 10.dp).size(40.dp),
+            text = sliderValue.toInt().toString()
+        )
```

Comment:
The code line 76-91 can be refactored as a new Composable component.  it has been used several times in the project.

Reply:
Applied the suggestion! Thank you.